rancher2/schema_cluster_rke_config_ingress.go

```diff
@@ -28,6 +28,11 @@ func clusterRKEConfigIngressFields() map[string]*schema.Schema {
 			Optional: true,
 			Computed: true,
 		},
+		"dns_policy": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
```

Comment:
What do you think to add argument validation to limit the value of `dns_policy` just to available at [K8S dns Policy](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy) ??

```
import (
	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
)

+const (
+	clusterRKEConfigIngressDNSPolicyClusterFirst = "ClusterFirst"
+	clusterRKEConfigIngressDNSPolicyClusterFirstWithHostNet = "ClusterFirstWithHostNet"
+	clusterRKEConfigIngressDNSPolicyDefault = "Default"
+	clusterRKEConfigIngressDNSPolicyNone = "None"
+)
+
+var (
+	clusterRKEConfigIngressDNSPolicyList = []string{
+		clusterRKEConfigIngressDNSPolicyClusterFirst, 
+		clusterRKEConfigIngressDNSPolicyClusterFirstWithHostNet,
+		clusterRKEConfigIngressDNSPolicyDefault,
+		clusterRKEConfigIngressDNSPolicyNone
+	}
+)
...
		"dns_policy": {
			Type:     schema.TypeString,
			Optional: true,
			Computed: true,
+			ValidateFunc: validation.StringInSlice(clusterRKEConfigIngressDNSPolicyList, true),
		},
```

Reply:
I thought about this but hesitated as tests on the Rancher API shows it accepts any string (even if this means the cluster fails as the dns policy is not recognised) - is it correct to add restrictions on the Terraform provider when there are no restrictions on Rancher? I'm for this change, but at the same time don't want to cause a limitation if there was a decision to make this an any-entry string on Rancher...

Reply:
The Rancher API accepts anything but RKE itself should not: https://github.com/rancher/rke/blob/master/cluster/validation.go#L171

Reply:
Changes added